### DISCUSSION TITLE
What's a good way to get the top left of the bounding rect from inside a node?

### DISCUSSION THREAD

**Discussion by raarts** [xyflow/xyflow#2411] [Q&A] [answered]
I think I need this, but here's my use case:

I am dragging a node over other nodes using the Drag & Drop API. I need to be able to see if I am dragging over the bottom half of the node or the top half. The event gives me clientX and clientY, and I should be able to project() that to the internal coordinate system, but for that I need to subtract the bounds top & left from the clientX and Y.

I can't find anything in the store (though it *does* have width and height of the bounding rect in pixels, why not x and y?) or any helper function.

I currently use a ref on the outer div to get the bounding rectangle, and put it into the data object of *every* node.
Maybe there's a better way?

**Comment by moklick** [comment, marked as answer]
Hey, `domNode` is part of the store:

`const bounds = useStore(s => s.domNode.getBoundingClientRect());`

**Reply by raarts** [reply]
Thanks! Never guessed this would be the value I needed!
